Sources/NetworkProtection/NetworkProtectionDeviceManager.swift

```diff
@@ -132,25 +132,11 @@ public actor NetworkProtectionDeviceManager: NetworkProtectionDeviceManagement {
                                             includedRoutes: [IPAddressRange],
                                             excludedRoutes: [IPAddressRange],
                                             isKillSwitchEnabled: Bool,
-                                            regenerateKey: Bool) async throws -> (TunnelConfiguration, NetworkProtectionServerInfo) {
-        var keyPair: KeyPair
+                                            keyPair: KeyPair) async throws -> (TunnelConfiguration, NetworkProtectionServerInfo) {
 
-        if regenerateKey {
-            keyPair = keyStore.newKeyPair()
-        } else {
-            keyPair = keyStore.currentKeyPair() ?? keyStore.newKeyPair()
-        }
-
-        let (selectedServer, newExpiration) = try await register(keyPair: keyPair, selectionMethod: selectionMethod)
+        let (selectedServer, _) = try await register(keyPair: keyPair, selectionMethod: selectionMethod)
         os_log("Server registration successul", log: .networkProtection)
 
-        keyStore.updateKeyPair(keyPair)
-
-        if let newExpiration {
-            keyPair = KeyPair(privateKey: keyPair.privateKey, expirationDate: newExpiration)
-            keyStore.updateKeyPair(keyPair)
-        }
```

Comment:
This part of the code was completely removed because it overrode the local 24hs expiration with the server's (1 year).  Do we need to add this back?

Reply:
The main thing currently puzzling me is that the old code _also_ did this - or at least, it appears to. See [this line](https://github.com/duckduckgo/BrowserServicesKit/pull/664/files#diff-d39b74340083f48d9124e45bc714ec990ab4a0b97fd668fd5afce6e695088ebeL217) in the diff where we used to update it.

We should confirm this behaviour with others before committing to it. Also, I would guess we want to add some migration code to update the expirations for those who have had it set months in the future to something closer, otherwise they'll continue to not rekey.